Tasks/DownloadPackageV1/package.ts

```diff
@@ -144,22 +145,22 @@ export abstract class Package {
     ): Promise<string> {
         const routeValues = {
             feedId: feedId,
-            project: project
+            project: project,
+            packageId: packageId
         };
         const queryParams = {
-            packageIdQuery: packageId,
-            protocolType: this.packageProtocolAreaName
+            isListed: true
         };
 
         return new Promise<string>(async (resolve, reject) => {
             this.getPackageMetadata(this.feedConnection, routeValues, queryParams, this.getPackagesAreaId)
-            .then(packages => {
-                tl.debug("Found " + packages["count"] + " packages matching search pattern " + packageId);
-                for (let i = 0; i < packages["count"]; i++) {
-                    if (packages["value"][i]["id"] == packageId && packages["value"][i]["versions"][0]["isListed"]) {
-                        return resolve(packages["value"][i]["versions"][0]["normalizedVersion"]);
+            .then(pkg => {
+                tl.debug("Found " + pkg["id"] + " package matching search pattern " + packageId);
+                pkg["versions"].forEach(version => {
```

Comment:
Can you also check for isListed. The latest version could be unlisted which means you cannot download it. 

It should be:
1. Check if latest version is listed, if yes download that.
2. If latest version is not listed, get the list of listed versions and select the first one of them

Reply:
Hi @aasim after reading the "isListed" document in the[ api](https://docs.microsoft.com/en-us/rest/api/azure/devops/artifacts/artifact-details/get-package?view=azure-devops-rest-7.1) I am not sure it would be good to include that check. It says:

"Only applicable for NuGet packages, setting it for other package types will result in a 404. If false, delisted package versions will be returned. Use this to filter the response when includeAllVersions is set to true. Default is unset (do not return delisted packages)."

Since the "includeAllVersions" is not a queryParam then it is not affected. Also this will not break on non-NuGet packages in this case.